### PR TITLE
ignore hidden columns

### PR DESCRIPTION
**Pull request by FrFerry**
When a column (POS, morph, lemma or similar) isn't displayed, then we don't export it and we don't take it into account for stats, history, ...

close: https://github.com/hipster-philology/pyrrha/issues/229

### CODE REVIEW

File: app/models/corpus.py [hunk 1]
```diff
@@ -701,7 +723,13 @@ def similar_as(cls, corpus: Corpus, form: str, lemma: str, POS: str, morph: str)
             cnt = 0
             for w in corpus.word_token:
                 if w.form == form:
-                    if w.lemma == lemma or w.POS == POS or w.morph == morph:
+                    if (
+                        "lemma" not in corpus.displayed_columns_by_name and w.lemma == lemma
```
Comment (MrGecko) [inline app/models/corpus.py:727]: Why ```not in``` ?

Comment (MrGecko) [inline app/models/corpus.py:727]: We may want to cache ```corpus.displayed_columns_by_name``` outside of this loop

**Comment by MrGecko** [comment]
Hello @FrFerry, two tests about exports are failing, could you please investigate?

**Comment by FrFerry** [comment]
> Hello @FrFerry, two tests about exports are failing, could you please investigate?

Yes, I think I've found why (some unwanted file reformatting)

**Comment by MrGecko** [comment]
@PonteIneptique should we still compute similars when there is no lemma, morph nor POS visible (it would means only forms are similar)?

**Comment by PonteIneptique** [comment]
Hmm, actually, I think it should not be possible to disable all the three annotation type...

**Comment by MrGecko** [comment]
> Hmm, actually, I think it should not be possible to disable all the three annotation type...

I've created the issue #239 to reference this new constraint.

**Comment by MrGecko** [comment]
Regarding the exports:
- In the CSV file, the hidden columns are still present but their value is now empty. I don't think this is what  we want. We may want to completly discard the hidden columns from the export.
- In the TEI file, the hidden columns are still rendered as empty lines in the XML file, this is an unwanted waste of space.
- in the TEI-geste file, the hidden columns are still rendered but empty : ```POS=|```

@PonteIneptique could you validate that hidden columns must not appear in the CSV header and in the TEI-geste file :) ?

**Comment by PonteIneptique** [comment]
I mean, if the column is empty, it's just a lost of size, but as long as there is the column it's not an issue. I would not think it's something to remove.
Agreed on TEI though.

**Comment by FrFerry** [comment]
I removed empty columns in CSV and also useless spaces in TEI.

**Comment by MrGecko** [comment]
I'm sorry @FrFerry, as stated by @PonteIneptique we decided to keep the empty columns in the CSV, could you please bring them back?
The other exports now look fine to me, thank you.

**Comment by PonteIneptique** [comment]
There is a conflict now because of #235

**Comment by MrGecko** [comment]
I will solve the conflict and merge this issue.

**Comment by FrFerry** [comment]
Should I bring empty columns back in the CSV ? (I received an email, but can't see the comment here)

**Comment by MrGecko** [comment]
> Should I bring empty columns back in the CSV ? (I received an email, but can't see the comment here)

No thank you, I'm about to solve all of this :)

**Review by MrGecko — APPROVED** [review]

**Pull request merged**